### PR TITLE
Release 1.8.1

### PR DESCRIPTION
**Pull request by MarioBassem**
## Changes

- Supports provider parallelism.
- Code cleanup.
- improve timeout handling in node client.
- Upgrade rmb-sdk-go version.

### CODE REVIEW

**Review by ashraffouda — APPROVED** [review]

File: internal/node/node.go [hunk 1]
```diff
@@ -117,6 +117,10 @@ type NodeClient struct {
 
 type args map[string]interface{}
```
Comment (xmonader) [inline internal/node/node.go:118]: Are these rmbCmdArgs?

File: internal/provider/validation.go [hunk 2]
```diff
@@ -36,22 +38,41 @@ func validateAccount(apiClient *apiClient, sub subi.SubstrateExt) error {
...
+	if !alphaOnly.MatchString(url) {
+		return errors.New("substrate url is not valid")
+	}
+
 	return nil
 }
 
-func preValidate(apiClient *apiClient, sub subi.SubstrateExt) error {
-	return validateRMBProxy(apiClient)
+func validateProxyURL(url string) error {
```
Comment (xmonader) [inline internal/provider/validation.go:62]: do we still need that for the relay? also the relay works on wss no?

File: internal/provider/validation.go [hunk 3]
```diff
@@ -36,22 +38,41 @@ func validateAccount(apiClient *apiClient, sub subi.SubstrateExt) error {
...
+	alphaOnly := regexp.MustCompile(`^https:\/\/[a-z0-9]+\.[a-z0-9]\/?([^\s<>\#%"\,\{\}\\|\\\^\[\]]+)?$`)
+	if !alphaOnly.MatchString(url) {
+		return errors.New("proxy url is not valid")
+	}
+
+	return nil
 }
 
-func validateAccountMoneyForExtrinsics(sub subi.SubstrateExt, identity substrate.Identity) error {
+func validateAccountBalanceForExtrinsics(sub subi.SubstrateExt, identity substrate.Identity) error {
```
Comment (xmonader) [inline internal/provider/validation.go:75]: please change in the logging money to balance

File: internal/provider/validation.go (hunk 2 above)
Comment (MarioBassem) [reply internal/provider/validation.go:62]: this validate the gridproxy's url, which is still used even after introducing the relay stuff

**Review by xmonader — APPROVED** [review]

**Pull request merged**